orchagent/orch.cpp

```diff
@@ -883,26 +896,44 @@ void Orch2::doTask(Consumer &consumer)
             }
             else
             {
-                SWSS_LOG_ERROR("Wrong operation. Check RequestParser: %s", op.c_str());
+                rc = ReturnCode(StatusCode::SWSS_RC_INVALID_PARAM)
+                     << "Wrong operation, only supporting SET and DEL, but got "
+                     << op;
+                SWSS_LOG_ERROR("%s", rc.message().c_str());
             }
         }
         catch (const std::invalid_argument& e)
         {
-            SWSS_LOG_ERROR("Parse error: %s", e.what());
+            rc = ReturnCode(StatusCode::SWSS_RC_INVALID_PARAM)
+                 << "Parse error: " << e.what();
+            SWSS_LOG_ERROR("%s", rc.message().c_str());
         }
         catch (const std::logic_error& e)
         {
-            SWSS_LOG_ERROR("Logic error: %s", e.what());
+            rc = ReturnCode(StatusCode::SWSS_RC_INTERNAL)
+                 << "Logic error: " << e.what();
+            SWSS_LOG_ERROR("%s", rc.message().c_str());
+            SWSS_RAISE_CRITICAL_STATE(rc.message());
```

Comment:
@mint570 - is this intended? Did we add the critical state API in?

Reply:
Fixed